ygen/codegen.go

```diff
@@ -653,21 +678,33 @@ func (cg *YANGCodeGenerator) GenerateProto3(yangFiles, includePaths []string) (*
 	if errs != nil {
 		return nil, errs
 	}
-	// If UseConsistentNamesForProtoUnionEnums=true, then also set
-	// appendEnumSuffixForSimpleUnionEnums=true for consistent union enum
-	// names.
-	enumSet, penums, errs := findEnumSet(mdef.enumEntries, cg.Config.TransformationOptions.CompressBehaviour.CompressEnabled(), true, cg.Config.ParseOptions.SkipEnumDeduplication, cg.Config.TransformationOptions.ShortenEnumLeafNames, cg.Config.TransformationOptions.UseDefiningModuleForTypedefEnumNames, cg.Config.ProtoOptions.UseConsistentNamesForProtoUnionEnums, cg.Config.ProtoOptions.UseConsistentNamesForProtoUnionEnums, cg.Config.TransformationOptions.EnumOrgPrefixesToTrim)
+
+	opts := IROptions{
+		ParseOptions:          cg.Config.ParseOptions,
+		TransformationOptions: cg.Config.TransformationOptions,
+		NestedDirectories:     true,
+		AbsoluteMapPaths:      true,
+		// If UseConsistentNamesForProtoUnionEnums=true, then also set
+		// appendEnumSuffixForSimpleUnionEnums=true for consistent union enum
+		// names.
+		AppendEnumSuffixForSimpleUnionEnums:  cg.Config.ProtoOptions.UseConsistentNamesForProtoUnionEnums,
+		UseConsistentNamesForProtoUnionEnums: cg.Config.ProtoOptions.UseConsistentNamesForProtoUnionEnums,
```

Comment:
Should we just make `UseConsistentNamesforProtoUnionEnums` always set `AppendEnumSuffixForSimpleUnionEnums` to be true -- and not really present this as two options? Is there any reason you would want inconsistently named enumerations?

Reply:
`AppendEnumSuffixForSimpleUnionEnums` fixes a version of inconsistent union names for Go generated code only. The fixes for proto enum naming was done all at once, so that's why it's both or neither: refs:

* https://github.com/openconfig/ygot/pull/474
* https://github.com/openconfig/ygot/pull/483

Reply:
ACK - understand that it is both or neither -- my suggestion was that if `UseConsistentNamesForProtoUnionEnums` is set, should we automatically set the other one? alternatively, should we error if they are inconsistently set?

Reply:
I guess for Go generation `UseConsistentNamesForProtoUnionEnums` is always false, so that's why for Go they could be different. If I made them the same field in `IROptions`, then within `GenerateIR` the call to `findEnumSet` won't be able to separate them from each other, and this would then mean the proto-specific logic cannot be switched off. Now it's possible that somehow we can merge the two flags but I think it's really not much of a problem for them to exist side-by-side for < 1 year before ygot@v1 is released, at which time they would be deleted.

Reply:
Ah, I see. OK, separate is OK with me, despite it being a bit confusing. I'd rather just get towards v1 when we can deprecate this :-)